api/desecapi/tests/test_rrsets.py

```diff
@@ -42,15 +44,90 @@ def test_retrieve_my_rr_sets(self):
+        self.assertEqual(links['first'].find('/?cursor='), links['next'].find('/?cursor='))
+        self.assertTrue(len(links['next']) > len(links['first']))
+
+        # Next
+        response = self.client.get(links['next'])
+        data_next = response.data.copy()
+        self.assertStatus(response, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), settings.REST_FRAMEWORK['PAGE_SIZE'])
+        links = convert_links(response['Link'])
+        self.assertEqual(len(links), 3)
+        self.assertTrue(links['first'].endswith('/?cursor='))
+        self.assertEqual(links['first'].find('/?cursor='), links['next'].find('/?cursor='))
+        self.assertTrue(len(links['next']) > len(links['first']))
+        self.assertEqual(links['first'].find('/?cursor='), links['prev'].find('/?cursor='))
+        self.assertTrue(len(links['prev']) > len(links['first']))
+
+        # Next-next (last) page
+        response = self.client.get(links['next'])
+        self.assertStatus(response, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), n/5 + 1)
+        links = convert_links(response['Link'])
+        self.assertEqual(len(links), 2)
+        self.assertTrue(links['first'].endswith('/?cursor='))
+        self.assertEqual(links['first'].find('/?cursor='), links['prev'].find('/?cursor='))
+        self.assertTrue(len(links['prev']) > len(links['first']))
+
+        # Prev
+        response = self.client.get(links['prev'])
+        self.assertStatus(response, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), settings.REST_FRAMEWORK['PAGE_SIZE'])
+        links = convert_links(response['Link'])
+        self.assertEqual(len(links), 3)
+        self.assertTrue(links['first'].endswith('/?cursor='))
+        self.assertEqual(links['first'].find('/?cursor='), links['next'].find('/?cursor='))
+        self.assertTrue(len(links['next']) > len(links['first']))
+        self.assertEqual(links['first'].find('/?cursor='), links['prev'].find('/?cursor='))
+        self.assertTrue(len(links['prev']) > len(links['first']))
+
+        # Make sure that one step forward equals two steps forward and one step back
+        self.assertEqual(response.data, data_next)
+
     def test_retrieve_other_rr_sets(self):
         self.assertStatus(self.client.get_rr_sets(self.other_domain.name), status.HTTP_404_NOT_FOUND)
         self.assertStatus(self.client.get_rr_sets(self.other_domain.name, subname='test'), status.HTTP_404_NOT_FOUND)
         self.assertStatus(self.client.get_rr_sets(self.other_domain.name, type='A'), status.HTTP_404_NOT_FOUND)
 
     def test_retrieve_my_rr_sets_filter(self):
-        response = self.client.get_rr_sets(self.my_rr_set_domain.name)
+        response = self.client.get_rr_sets(self.my_rr_set_domain.name, query='?cursor=')
         self.assertStatus(response, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), len(self._test_rr_sets()))
```

Comment:
can we assert that length is `min(len(self...), page_size_setting)`?

Reply:
sure! good solution, done